File: eculid/euclid.c
/************************/
/*  Euclidean Alogrithm */
/*   by Conor Gilmer    */
/*      euclid.c        */
/*      circa 1995      */
/************************/
#include<stdio.h>

int gcd_one (int, int);
int gcd_two (int, int);

int main()
{
	int x,y;

	printf("\n\t*** Euclidian Algorithm *** ");
	printf("\n\t*** Find the Greatest Common Divisor *** ");
	printf("\n\tEnter first number : ");
	scanf("%d",&x);
	printf("\tEnter second number: ");
	scanf("%d",&y);
	printf("\n\tGreatest Common Divisor for (%d, %d) is %d", x, y,  gcd_one(x,y));
	printf("\n\tGreatest Common Divisor for (%d, %d) is %d", x, y,  gcd_two(x,y));
	printf("\n\t*** The End *** \n");
	
} /* end of main */


/* gcd using while */
int gcd_one(int p, int q) {
        while (q != 0) {
            int temp = q;
            q = p % q;
            p = temp;
        }
        return p;
    } /* end of gcd one */


/* gcd using do while */
int gcd_two (int u, int v)
{
int t;
	do {
		if (u < v)
		{
			t = u;
			u = v;
			v = t;
		}
		u = u - v;
	} while (u != v);

return u;
} /* end of gcd two */

File: euclid/Euclid.java
/******************************************************************************
 *  Program:  javac Euclid.java
 *  Run:      java Euclid p q
 *  
 *  Calculates the Greatest Common Divisor (GCD) using Euclids Algorithm.
 *  Three different functions are shown here how to calculate it  
 * 
 *
 ******************************************************************************/

public class Euclid {

    /* a recursive gcd implementation */
    public static int gcd_rec(int p, int q) {
        if (q == 0) 
	    return p;
        else 
	    return gcd_rec(q, p % q);
    } /* end of gcd rec */

    /* a non-recursive gcd implementation using while */
    public static int gcd_one(int p, int q) {
        int t;
        while (q != 0) {
            t = q;
            q = p % q;
            p = t;
        }
        return p;
    } /* end of gcd one */


    /* a non-recursive gcd using do-while */
    public static int gcd_two (int p, int q)
    {
    int t;
	do {
		if (p < q)
		{	/* swap them */
			t = p;
			p = q;
			q = t;
		}
		p = p - q;
	} while (p != q);

    return p;
    } /* end of gcd two */


    /* main function */
    public static void main(String[] args) {

	// get inputs
        int p     = Integer.parseInt(args[0]);
        int q     = Integer.parseInt(args[1]);

	// call gcd functions
        int dRec  = gcd_rec(p, q);
        int dOne  = gcd_one(p, q);
        int dTwo  = gcd_two(p, q);

        System.out.println("*** Greatest Common Divisor (GCD) ***");
	// print out results (they should all be the same hopefully).
        System.out.println("GCD (" + p + ", " + q + ") = " + dRec);
        System.out.println("GCD (" + p + ", " + q + ") = " + dOne);
        System.out.println("GCD (" + p + ", " + q + ") = " + dTwo);

        System.out.println("*** The End ***");
    } /* end of main function */

} /* end of Euclid class */


